### PR TITLE
Update spellchecking whitelist and fix more typos

### PR DESCRIPTION
**Pull request by msimberg**
This fixes more typos and expands the whitelist. I've also changed the CircleCI config to not require `spellcheck`, `inspect`, and `docs` for the `install` step (as they don't affect functionality).

### CODE REVIEW

File: examples/transpose/transpose_serial.cpp [hunk 1]
```diff
@@ -152,8 +152,8 @@ int main(int argc, char* argv[])
         ( "verbose", "Verbose output")
     ;
 
-    // Initialize and run HPX, this example is serial and therefor only needs on thread,
-    // We just use hpx::init to parse our command line arguments
+    // Initialize and run HPX, this example is serial and therefore only needs
+    // on thread, We just use hpx::init to parse our command line arguments
```
Comment (hkaiser) [inline examples/transpose/transpose_serial.cpp]: Should it be `one thread` instead?

File: examples/transpose/transpose_serial_vector.cpp [hunk 2]
```diff
@@ -157,8 +157,8 @@ int main(int argc, char* argv[])
         ( "verbose", "Verbose output")
     ;
 
-    // Initialize and run HPX, this example is serial and therefor only needs on thread,
-    // We just use hpx::init to parse our command line arguments
+    // Initialize and run HPX, this example is serial and therefore only needs
+    // on thread, We just use hpx::init to parse our command line arguments
```
Comment (hkaiser) [inline examples/transpose/transpose_serial_vector.cpp]: Same here: `one thread`.

File: examples/transpose/transpose_serial.cpp (hunk 1 above)
Comment (msimberg) [reply examples/transpose/transpose_serial.cpp]: It helps to read around the typo as well... thanks!

**Comment by aurianer** [comment]
Thanks!

**Pull request merged**